src/main/java/net/minecraftforge/common/world/WorldGenValidator.java

```diff
@@ -0,0 +1,121 @@
+                    if (!checkedElements.contains(carver)) {
+                        validate(carver, configuredCarverRegistry, "ConfiguredCarver", ConfiguredWorldCarver.DIRECT_CODEC, biomeName);
+                        checkedElements.add(carver);
+                    }
+                });
+            }
+
+            biomeGenerationSettings.structures().forEach(structureFeatureSupplier -> {
+                ConfiguredStructureFeature<?, ?> structureFeature = structureFeatureSupplier.get();
+                if (!checkedElements.contains(structureFeature)) {
+                    validate(structureFeature, configuredStructureRegistry, "ConfiguredStructure", ConfiguredStructureFeature.DIRECT_CODEC, biomeName);
+                    checkedElements.add(structureFeature);
+                }
+            });
+        }
+    }
+
+    /**
+     * Handles checking if the worldgen element is broken (unable to be turned into JSON) or unregistered.
+     * If an issue is found, this will pause the IDE so the modder can come here and see the error and the worldgenElement for which element is the issue.
+     * The console output will have the Forge message as last thing when paused too.
+     */
+    private static <T> void validate(T worldgenElement, Registry<T> registry, String worldgenElementType, Codec<T> codec, ResourceLocation biomeName) {
+
+        // Checks to make sure the element can be turned into JSON safely or else Minecraft will explode with vague errors.
+        JsonElement worldgenElementJSON = codec.encode(worldgenElement, JsonOps.INSTANCE, JsonOps.INSTANCE.empty())
+                .getOrThrow(false, errorMsg ->
+                        LOGGER.error(worldgenElementType + " was unable to be turned into json in " + biomeName + " biome. Please breakpoint this line and look at worldgenElement above to see which worldgen element is broken. " +
+                                "Check to make sure your codecs are correct or it and that the values you give it does not exceed any limits in the object's codecs. Error: " + errorMsg));
+
+        // Checks to make sure the element is registered.
+        if (!registry.getResourceKey(worldgenElement).isPresent()) {
+            throw new UnsupportedOperationException(
+                    "Unregistered " + worldgenElementType + " found in " + biomeName + " biome. " +
+                            "Please register your " + worldgenElementType + " using Minecraft's Registry class within FMLCommonSetupEvent's enqueueWork. " +
+                            "Failure to do so may cause unregistered worldgen elements to nuke registered ones from the same biome which is difficult for players to debug. " +
+                            "See this json of the unregistered worldgen element to know which one is unregister: \n" + (worldgenElementJSON != null ? gson.toJson(worldgenElementJSON) : "Unable to be parsed into JSON"));
```

Comment:
This can possibly be moved to use text blocks and `String#formatted(Object...)`: `"""Unregisterd %s found in %s ...""".formatted(worldgenElementType, ...)`.